hack/lint.sh

```
@@ -2,14 +2,14 @@
 
 set -euo pipefail
 
-# pin golangci-lint version to 1.54.2
-VERSION=v1.54.2
+# pin golangci-lint version
+VERSION=v2.12.2
 if [ "$#" -ne 1 ]; then
     echo "Expected command line argument - container runtime (docker/podman) got $# arguments: $@"
```

Shellcheck (0.11.0)

[error] Argument mixes string and array. Use * or separate argument.

(SC2145)